File: app/src/main/java/knf/kuma/custom/HiddenOverlay.java
package knf.kuma.custom;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;
import butterknife.ButterKnife;
import knf.kuma.R;

public class HiddenOverlay extends LinearLayout {
    public HiddenOverlay(Context context) {
        super(context);
        inflate(context);
    }

    public HiddenOverlay(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        inflate(context);
    }

    public HiddenOverlay(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        inflate(context);
    }

    private void inflate(Context context) {
        LayoutInflater inflater = (LayoutInflater) context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        inflater.inflate(R.layout.view_hidden_overlay, this);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        ButterKnife.bind(this);
    }

    public void setHidden(final boolean hidden, boolean animate) {
        setState(hidden);
        if (animate)
            post(new Runnable() {
                @Override
                public void run() {
                    Animation animation = AnimationUtils.loadAnimation(getContext(), hidden ? R.anim.fadein : R.anim.fadeout);
                    animation.setDuration(200);
                    animation.setAnimationListener(new Animation.AnimationListener() {
                        @Override
                        public void onAnimationStart(Animation animation) {

                        }

                        @Override
                        public void onAnimationEnd(Animation animation) {

                        }

                        @Override
                        public void onAnimationRepeat(Animation animation) {

                        }
                    });
                    startAnimation(animation);
                }
            });
    }

    private void setState(final boolean hidden) {
        post(new Runnable() {
            @Override
            public void run() {
                setVisibility(hidden ? VISIBLE : GONE);
            }
        });
    }
}

File: app/src/main/java/knf/kuma/custom/HiddenOverlay.kt
package knf.kuma.custom

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import android.widget.LinearLayout
import knf.kuma.R

class HiddenOverlay : LinearLayout {
    constructor(context: Context) : super(context) {
        inflate(context)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        inflate(context)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        inflate(context)
    }

    private fun inflate(context: Context) {
        val inflater = context
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        inflater.inflate(R.layout.view_hidden_overlay, this)
    }

    fun setHidden(hidden: Boolean, animate: Boolean) {
        setState(hidden)
        if (animate)
            post {
                val animation = AnimationUtils.loadAnimation(context, if (hidden) R.anim.fadein else R.anim.fadeout)
                animation.duration = 200
                animation.setAnimationListener(object : Animation.AnimationListener {
                    override fun onAnimationStart(animation: Animation) {

                    }

                    override fun onAnimationEnd(animation: Animation) {

                    }

                    override fun onAnimationRepeat(animation: Animation) {

                    }
                })
                startAnimation(animation)
            }
    }

    private fun setState(hidden: Boolean) {
        post { visibility = if (hidden) View.VISIBLE else View.GONE }
    }
}
